File: source/FLDN_API/Domain/Entities/Product/Product.cs
namespace Domain;

public class Product : EntityBase<Guid>, ISoftDeletable
{
    public Guid SupplierId { get; set; }
    public Guid CategoryId { get; set; }
    public string Name { get; set; } = default!;
    public string? Description { get; set; }
    public decimal WholesalePrice { get; set; }
    public decimal RetailPrice { get; set; }
    public string Unit { get; set; } = default!;
    public string? PackagingStandard { get; set; }
    public bool IsActive { get; set; }
    public bool IsDeleted { get; set; }
    public DateTimeOffset? DeletedAt { get; set; }

    // Navigation
    public SupplierProfile Supplier { get; set; } = default!;
    public Category Category { get; set; } = default!;
    public ICollection<ProductImage> ProductImages { get; set; } = [];
    public ICollection<PriceHistory> PriceHistories { get; set; } = [];
    public Inventory? Inventory { get; set; }
    public ICollection<Batch> Batches { get; set; } = [];
}

File: source/FLDN_fe/src/types/product.ts
export interface Category {
  categoryId: string
  name: string
  description?: string
  imageUrl?: string
  parentCategoryId?: string | null
  isActive?: boolean
  children?: Category[]
}

export interface ProductImage {
  imageId: string
  imageUrl: string
  isMain: boolean
}

export interface Product {
  productId: string
  name: string
  description?: string
  retailPrice: number
  wholesalePrice?: number
  unit: string
  packagingStandard?: string
  isActive?: boolean
  supplierName?: string
  supplierId?: string
  mainImage?: string
  images?: ProductImage[]
  categoryName?: string
  categoryId?: string
  quantity?: number // stock
  averageRating?: number
  totalReviews?: number
  season?: string
  districtName?: string
}

export interface ProductSearchParams {
  keyword?: string
  categoryId?: string
  supplierId?: string
  minPrice?: number
  maxPrice?: number
  sort?: 'price_asc' | 'price_desc' | 'newest' | 'name_asc'
  page?: number
  pageSize?: number
}

export interface ProductSearchResponse {
  items: Product[]
  totalCount: number
  page: number
  pageSize: number
  totalPages: number
}
